diff --git a/Makefile.in b/Makefile.in
index 1eef3a5..dc1dfce 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -37,10 +37,10 @@ MODULE_INSTALL=${MODULES:%=%_inst}
 GPROPTS=-XXMLADA_BUILD_MODE=${MODE} -XPROCESSORS=${PROCESSORS}
 
 ifeq (${HOST},${TARGET})
-IPREFIX=${prefix}
+IPREFIX=${DESTDIR}/${prefix}
 else
 GPROPTS+=--target=${TARGET_ALIAS}
-IPREFIX=${prefix}/${TARGET_ALIAS}
+IPREFIX=${DESTDIR}/${prefix}/${TARGET_ALIAS}
 endif
 
 ifdef RTS
